File: Codes/Hash/Non-Cryptographic/BPHash/BPHash.cs
/*
	Hash function by Bruno Preiss
    Archive of Reversing.ID
    Non-Cryptographic Hash
*/

// Fungsi ini merupakan method dari sebuah class
public static uint BPHash (string key)
{
    uint state = 0;
    uint i;

    for (i = 0; i < key.Length; i++)
    {
        state = state << 7 ^ ((byte)key[(int)i];
    }
    return state;
}

File: Codes/Hash/Non-Cryptographic/BPHash/BPHash.pas
{
    Hash function by Bruno Preiss
    Archive of Reversing.ID
    Non-Cryptographic Hash
}
function BPHash(const key : String) : Cardinal;
var
    i : Cardinal;
begin
    Result := 0;

    for i := 1 to Length(key) do
    begin
        Result := Result shl 7 xor Ord(key[i]);
    end;
end;